dvc/hash_info.py

```diff
@@ -14,6 +14,7 @@ class HashInfo:
     value: Optional[str]
     size: Optional[int] = field(default=None, compare=False)
     nfiles: Optional[int] = field(default=None, compare=False)
+    obj_name: Optional[str] = field(default=None, compare=False)
```

Comment:
Hm, looks like `obj` is leaking here. I see that you only use it for status (prev obj.name), maybe there is a nicer way to handle that without leaking obj things into HashInfo?

Reply:
This is the remnants of NamedCache - to keep the current `dvc status -c` behavior (and output workspace path names instead of hashes) we have to return something during used object collection that maps objects/oids to path names. 

We could collect an additional dictionary that maps oids to names (so used_objs would return tuples of `(odb, oids, oid_names)`) so that `dvc status` can do the hash->name lookup, but at this point, `oid_names` would be a hack that only exists for legacy UI purposes (and is pretty much just the old `NamedCache`).

IMO, since `status -c`/`push`/`pull` operate on object IDs and don't actually have anything to do with "workspace files" at all, the nicest way to handle it would be to get rid of the name thing entirely, and change `status -c` so that it just outputs counts and size information for "total objects that would be uploaded, downloaded, or are missing" rather than the current behavior of listing workspace filenames for every object that would be uploaded/downloaded.

(But this would obviously be a significant behavior change for `dvc status -c`)

Reply:
@pmrowla Makes sense! Then we could keep obj_name for now and reconsider it in the near future when we'll be revisiting push/pull/status ui. 👍